src/aof.c

```diff
@@ -160,15 +186,135 @@ void aof_background_fsync(int fd) {
+        if ((server.aof_plp_rewr = pmemlog_open(server.aof_filename_rewr)) == NULL) {
+            redisLog(REDIS_WARNING, "Redis cannot open NVML AOF REWR file: %s", strerror(errno));
+            exit(1);
+        }
+    } else {
+        if ((server.aof_plp_rewr = pmemlog_create(server.aof_filename_rewr, server.aof_nvml_log_size, 0600)) == NULL) {
+            redisLog(REDIS_WARNING, "Redis cannot create NVML AOF REWR file: %s", strerror(errno));
+            return REDIS_ERR;
+        }
+    }
+
+    pmemlog_rewind(server.aof_plp_rewr);
+
+    return REDIS_OK;
+}
+#endif
+
+void openAppendOnlyFile(void)
+{
+    if (server.aof_fd != -1) {
+        return;
+    }
+#ifdef USE_NVML
+    if (server.aof_use_nvml == 1) {
+        /* First open regular AOF PMEMLOG pool, then pool for rewrite
+         * and if size of loaded file is different, than configured then
+         * trigger rewrite */
+
+        int stat_ret;
+        struct stat *stat_buf;
+
+        /* if file doesn't exist create it and truncate to configured size */
+        stat_buf = zmalloc(sizeof (struct stat));
+        stat_ret = stat(server.aof_filename, stat_buf);
+        if (stat_ret != 0) {
+            if (errno != ENOENT) {
+                redisLog(REDIS_WARNING, "Can't stat the AOF file: %s",
+                         strerror(errno));
+                exit(1);
+            } else {
+                /* file don't exist: create it */
+                if ((server.aof_plp = pmemlog_create(server.aof_filename, server.aof_nvml_log_size, 0600)) == NULL) {
+                    redisLog(REDIS_WARNING, "Redis cannot create NVML AOF file: %s", strerror(errno));
+                    exit(1);
+                }
+            }
+        } else {
+            /* file exists: open it */
+            if ((server.aof_plp = pmemlog_open(server.aof_filename)) == NULL) {
+                redisLog(REDIS_WARNING, "Redis cannot open NVML AOF file: %s", strerror(errno));
+                exit(1);
+            }
+        }
+        zfree(stat_buf);
+
+        /* For redis non-negaitve aof_fd indicates, that openning aof file
+         * succeeded. PMEMlog handles fd internally, so here aof_fd is set to
+         * non-negative value just to don't break redis behaviour. */
+        server.aof_fd = 1;
+        openAppendOnlyFileREWR();
+
+        aofUpdateCurrentSize();
+    } else {
+#endif
+        /* Regular AOF file */
+        server.aof_fd = open(server.aof_filename, O_WRONLY|O_APPEND|O_CREAT,0644);
+        if (server.aof_fd == -1) {
+            redisLog(REDIS_WARNING, "Can't open the append-only file: %s", strerror(errno));
+            exit(1);
+        }
+#ifdef USE_NVML
+    }
+#endif
+}
+
+#ifdef USE_NVML
+void closeAppendOnlyFileREWR(void) {
+    if (server.aof_plp_rewr != NULL) {
+        pmemlog_close(server.aof_plp_rewr);
+        server.aof_plp_rewr = NULL;
+    }
+}
+#endif
+
+int closeAppendOnlyFile(void)
+{
+#ifdef USE_NVML
+    if (server.aof_use_nvml == 1
```

Comment:
If `aof_use_nvml` is not 1, then `aof_plp` should be NULL, I guess. So, what about checking only `aof_plp`?